native/c/zds.cpp

```diff
@@ -748,6 +750,459 @@ void load_recfm_from_dscb(const DSCBFormat1 *dscb, string *recfm)
+void load_general_attrs_from_dscb(const DSCBFormat1 *dscb, ZDSEntry &entry)
+{
+  load_dsorg_from_dscb(dscb, &entry.dsorg);
+
+  // Determine space unit first (needed for other calculations)
+  load_spacu_from_dscb(dscb, entry);
+
+  // Check if this is a VSAM dataset
+  bool is_vsam = (dscb->ds1dsorg & DS1ACBM_MASK) != 0;
+
+  if (is_vsam)
+  {
+    // VSAM datasets: blksize, lrecl, and recfm are not meaningful
+    entry.blksize = -1;
+    entry.lrecl = -1;
+    entry.recfm = "";
+  }
+  else
+  {
+    load_recfm_from_dscb(dscb, &entry.recfm);
+    entry.blksize = (static_cast<unsigned char>(dscb->ds1blkl[0]) << 8) |
+                    static_cast<unsigned char>(dscb->ds1blkl[1]);
+    entry.lrecl = (static_cast<unsigned char>(dscb->ds1lrecl[0]) << 8) |
+                  static_cast<unsigned char>(dscb->ds1lrecl[1]);
+  }
+
+  // Determine dsntype: PDS or LIBRARY (PDSE)
+  // DS1DSGPO (0x0200) in ds1dsorg indicates partitioned organization
+  bool is_partitioned = (dscb->ds1dsorg & DS1DSGPO_MASK) != 0;
+  // DS1PDSE (0x10) in ds1smsfg indicates PDSE
+  bool is_pdse = (dscb->ds1smsfg & DS1PDSE_MASK) != 0;
+
+  if (is_partitioned && is_pdse)
+  {
+    entry.dsntype = "LIBRARY";
+  }
+  else if (is_partitioned)
+  {
+    entry.dsntype = "PDS";
+  }
+  else
+  {
+    entry.dsntype = "";
+  }
+
+  // Check if dataset is encrypted
+  // DS1ENCRP (0x04) in ds1flag1 indicates access method encrypted dataset
+  entry.encrypted = (dscb->ds1flag1 & DS1ENCRP_MASK) != 0;
+
+  // Load primary and secondary allocations (part of General Data in ISPF)
+  load_primary_secondary_from_dscb(dscb, entry);
+}
+
+// Load Current Allocation attributes (allocated units/tracks/bytes, allocated extents)
+void load_alloc_attrs_from_dscb(const DSCBFormat1 *dscb, ZDSEntry &entry)
+{
+  uint8_t scal1 = static_cast<uint8_t>(dscb->ds1scalo[0]);
+  uint8_t dspac = (scal1 & 0xC0) >> 6;
+
+  // allocx: Number of extents allocated on this volume
+  entry.allocx = dscb->ds1noepv;
+
+  // Parse the extent information from ds1exnts (3 extents, 10 bytes each)
+  const char *extent_data = dscb->ds1exnts;
+  int total_cylinders = 0;
+  int total_tracks = 0;
+  bool use_cylinders = (dspac == 0x03);
+
+  for (int i = 0; i < 3; i++)
+  {
+    const char *extent = extent_data + (i * 10);
+    uint8_t extent_type = static_cast<uint8_t>(extent[0]);
+
+    if (extent_type == 0x00)
+      break;
+
+    uint32_t lower_cyl = parse_cchh_cylinder(extent + 2);
+    uint32_t upper_cyl = parse_cchh_cylinder(extent + 6);
+
+    if (upper_cyl >= lower_cyl)
+    {
+      int cylinders_in_extent = upper_cyl - lower_cyl + 1;
+      int tracks_in_extent = calculate_extent_tracks(extent);
+      total_cylinders += cylinders_in_extent;
+      total_tracks += tracks_in_extent;
+    }
+  }
+
+  entry.alloc = use_cylinders ? total_cylinders : total_tracks;
+
+  // Convert to bytes if spacu is BYTES
+  if (entry.spacu == "BYTES" && entry.blksize > 0)
+  {
+    int blocks_per_track = 56664 / entry.blksize;
```

Comment:
Q: `56664` is the track capacity for 3390 devices, but doesn't this vary by device type? We are parsing the `devtype` below, but here it seems we are assuming 3390 volume for space calculation

Also, thoughts on moving this "magic number" to a constant variable if we can rely on this single value?